lib/nats.js

```diff
@@ -401,7 +444,7 @@ Client.prototype.setupHandlers = function() {
     // Only bubble up error if we never had connected
     // to the server and we only have one.
     if (client.wasConnected === false && client.servers.length === 0) {
-      client.emit('error', 'Could not connect to server: ' + exception);
+      client.emit('error', new NatsError(CONN_ERR_MSG_PREFIX + exception, CONN_ERR, exception));
```

Comment:
OK I see just thought that maybe keeping the original type could be useful, but maybe better if the error types are uniform as you mention.